mantidproject/mantid
# Updated mantidworkbench to send stacktrace

**Description of work.**

This updates the error reporter in two ways
  * It removes the sending of recovery files as these were found to be not very useful.
  * It adds in the sending of a stacktrace in workbench in cases where one can be retrieved.

When this is merged we should update the [privacy policy](https://www.mantidproject.org/MantidProject:Privacy_policy#Usage_Data_recorded_in_Mantid) to reflect what is now sent.



**To test:**
  * To test this properly you will need to get a local version of the error report server with this pull request built. https://github.com/mantidproject/errorreports/pull/15
  * You will also need to change the errorreports url in Mantid.properties.template to http://localhost:8082
  * There are two cases which need to be tested. 
       * Testing the catching of non-crashing errors can be done by trying to cancel a mantid build via a keyboard interrupt. 
       * To test the other case you need to add `raise RuntimeError("Error Message") before line 859 in mainwindow.py . This will then provoke an excetion on startip.

  * Check that in both cases check that error-reports are received by the server with appropriate stack traces.




*There is no associated issue.*



*This does not require release notes* because this is a developer facing change


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
